File: src/main/java/com/randomperson22/piggymodtest/npc/georgie/EntityGeorgieNpc.java
package com.randomperson22.piggymodtest.npc.georgie;

import com.randomperson22.piggymodtest.npc.EntityNpc;
import net.minecraft.init.Items;
import net.minecraft.item.Item;
import net.minecraft.world.World;

public class EntityGeorgieNpc extends EntityNpc {
    public EntityGeorgieNpc(World worldIn) {
        super(worldIn);
    }

    @Override
    protected Item getRequiredItem() {
        return Items.APPLE;
    }
    
    @Override
    public void onUpdate() {
        super.onUpdate();
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/npc/georgie/EntityGeorgieNpc.kt
package com.randomperson22.piggymodtest.npc.georgie

import com.randomperson22.piggymodtest.npc.EntityNpc
import net.minecraft.init.Items
import net.minecraft.item.Item
import net.minecraft.world.World

class EntityGeorgieNpc(worldIn: World) : EntityNpc(worldIn) {
    override fun getRequiredItem(): Item? {
        return Items.APPLE
    }

    override fun onUpdate() {
        super.onUpdate()
    }
}